sdv/metadata/table.py

```diff
@@ -553,6 +554,25 @@ def _transform_constraints(self, data, on_missing_column='error'):
 
         return data
 
+    def _validate_data_on_constraints(self, data):
+        """Make sure the given data is valid for the given constraints.
+
+        Args:
+            data (pandas.DataFrame):
+                Table data.
+
+        Returns:
+            bool:
+                Whether or not the table data is valid for the constraints.
+        """
+        is_valid = True
+
+        for constraint in self._constraints:
+            if set(constraint.constraint_columns).issubset(data.columns.values):
+                is_valid = is_valid and constraint.is_valid(data).all()
```

Comment:
What about raising the exception directly from here, so we do not need to continue validating?

Reply:
So I personally prefer raising it in `transform`. I feel like it scopes this method to answering the question, "is it valid or not?", and then the caller can determine what to do if not valid.

I can make it return early if any of the `constraint.is_valid` is False though, since it doesn't need to keep iterating in that case.

Reply:
Update: I moved the error to the helper method, so that it is thrown immediately